ledger/common/address.go

```diff
@@ -53,15 +54,25 @@ type Address struct {
 	extraData      []byte
 }
 
-// NewAddress returns an Address based on the provided bech32 address string
+// NewAddress returns an Address based on the provided bech32/base58 address string
+// It detects if the string has mixed case assumes it is a base58 encoded address
+// otherwise, it assumes it is bech32 encoded
 func NewAddress(addr string) (Address, error) {
-	_, data, err := bech32.DecodeNoLimit(addr)
-	if err != nil {
-		return Address{}, err
-	}
-	decoded, err := bech32.ConvertBits(data, 5, 8, false)
-	if err != nil {
-		return Address{}, err
+	var decoded []byte
+	var err error
+
+	if isMixedCase(addr) {
```

Comment:
We can probably drop the helper function and just do something like `if strings.ToLower(addr) != addr {`. This will effectively detect upper-case characters in the string, which we should never see for bech32.

Reply:
That's a smart move, thanks